src/abstract-interpretation/ainode.ts

```diff
@@ -0,0 +1,48 @@
+import { NodeId, ParentInformation, RNodeWithParent } from '../r-bridge'
+import { assertUnreachable, guard } from '../util/assert'
+import { Domain, unifyDomains } from './domain'
+
+export interface AINode {
+	// The ID of the node that logically holds the domain
+	readonly nodeId:       NodeId
```

Comment:
why not join this with the respective astNode? as they are inherently linked by the traversal it could be a direct decoration.

Reply:
Addressed in e6d3add0